secure/storage/src/vault.rs

```diff
@@ -304,9 +306,9 @@ impl CryptoStorage for VaultStorage {
     ) -> Result<Ed25519Signature, Error> {
         let name = self.crypto_name(name);
         let mut bytes = <T::Hasher as libra_crypto::hash::CryptoHasher>::seed().to_vec();
-        lcs::serialize_into(&mut bytes, &message)
-            .map_err(|_| libra_crypto::traits::CryptoMaterialError::SerializationError)
-            .expect("Serialization of signable material should not fail.");
```

Comment:
I think this is an expect, why would it fail? If it fails, I want the system to tell me and not make further progress.

Reply:
why don't you log a MUST_FIX instead so that you can continue to perform the other operations that still work? Also maybe you will retry with a different message and it'll work. No reason to panic if we can work on the fix while this continues to run : o

Reply:
It is my understanding that:
- we do not have, either in signing or secure storage, comprehensive pre-execution checks of recursion depths, in the way that the language / VM does (#4993 #5023)¹
- yet signables are very much under the same serialization limit as everybody else (#4928), which makes at least one plausible reason where serialization could fail,
- I don't know for sure if we have any guarantees that at this point of call the contents message is controlled by the current process (rather than external input),
- it's my understanding this panic will at least irrecoverably blow up the secure storage service, making the node fail and become byzantine.

This seems to look — and please correct me if I'm wrong — like a textbook vulnerability: any attacker able to make generate a value that will fail serialization knocks down the node.

On the other hand, could you remind me of the issues that would be created in not managing to sign this particular message, and trying to move on to the next one nonetheless? 

¹: are they planned? have I missed something? /cc @matbd

Reply:
what is the internal error? lost context.